frontend/src/main/components/Courses/CoursesTable.js

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+ import OurTable, { ButtonColumn } from "main/components/OurTable"
+ import { useBackendMutation } from "main/utils/useBackend";
+ import { cellToAxiosParamsDelete, onDeleteSuccess } from "main/utils/coursesUtils";
+ import { useNavigate } from "react-router-dom";
+ import { hasRole } from "main/utils/currentUser";
+
+ export default function CoursesTable({ courses, currentUser }) {
+
+     const navigate = useNavigate();
+
+     const joinCallback = (cell) => {
+        navigate(`/courses/join/${cell.row.values.id}`);
+    };
```

Comment:
What is this 'join' page?   I don't think we have one in this app.  If I'm mistaken, please explain.
```suggestion

```